src/cryptonote_core/tx_pool.cpp

```diff
@@ -269,6 +269,14 @@ bool tx_memory_pool::have_duplicated_non_standard_tx(
                 return true;
             }
         }
+
+        std::unique_lock b_lock{m_blockchain};
```

Comment:
Do we know where these TXs are coming from? Could be the P2P layer if it's from a malicious node, but in normal operations ... How does this get triggered?